main.tf

```diff
@@ -0,0 +1,98 @@
+terraform {
+  required_version = ">= 0.13"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 2.84.0"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+locals {
+
+  diag_resource_list = var.diagnostics != null ? split("/", var.diagnostics.destination) : []
+  parsed_diag = var.diagnostics != null ? {
+    log_analytics_id   = contains(local.diag_resource_list, "Microsoft.OperationalInsights") ? var.diagnostics.destination : null
+    storage_account_id = contains(local.diag_resource_list, "Microsoft.Storage") ? var.diagnostics.destination : null
+    event_hub_auth_id  = contains(local.diag_resource_list, "Microsoft.EventHub") ? var.diagnostics.destination : null
+    metric             = var.diagnostics.metrics
+    log                = var.diagnostics.logs
+    } : {
+    log_analytics_id   = null
+    storage_account_id = null
+    event_hub_auth_id  = null
+    metric             = []
+    log                = []
+  }
+}
+
+resource "azurerm_resource_group" "main" {
+  name     = var.resource_group_name
+  location = var.location
+
+  tags = var.tags
+}
+
+resource "azurerm_cognitive_account" "main" {
+  for_each = var.cognitive_services_types
+
+  name                = "${each.value.kind}-cog"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+  kind                = each.value.kind
+  sku_name            = each.value.sku_name
+  tags                = var.tags
+}
```

Comment:
Add network_acls. Should any other attributes be added as well ?

Reply:
custom_subdomain_name is required if network_acls is used even though it is specified as optional. Issue registered here https://github.com/hashicorp/terraform-provider-azurerm/issues/10041